### PR TITLE
docs: Add group module docs

### PR DESCRIPTION
**Pull request by glnro**
This pr fulfills the documentation component of #1423

### CODE REVIEW

File: docs/modules/group.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+## Overview
+A group can be thought of as a set of accounts that can have multiple multi-signature accounts in the form of decision policy. Furthermore, each decision policy can have a different threshold for voting on and passing proposals.
+
+See the following resources for additional information on using the `group` module.
+
+- [Cosmos SDK Group Module Docs](https://docs.cosmos.network/v0.46/modules/group/)
+- [Cosmos Developer Portal Group Tutorial](https://tutorials.cosmos.network/tutorials/understanding-group/)
+
+##### Group Administration
+Groups can be initialized with or without a decision policy. Groups must have an administrator account, but that account does not necessarily have to be a group member. Upon creating a group, each member is assigned a weight that will correspond to voting power with respect to policy proposals.
```
Comment (hxrts) [inline docs/modules/group.md]:
```suggestion
Groups can be initialized with or without a decision policy. Groups must have an administrator account, but that account does not necessarily have to be a group member. The group may self-administer by setting the administrator to a group policy account. Upon creating a group, each member is assigned a weight that will correspond to voting power with respect to policy proposals.
```

File: docs/modules/group.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# Group Module
+
+**The `group` module facilitates the creation and management of multisig accounts and enables members to configure group policies that can submit and vote on specific message execution proposals.**
+
+## Overview
+A group can be thought of as a set of accounts that can have multiple multi-signature accounts in the form of decision policy. Furthermore, each decision policy can have a different threshold for voting on and passing proposals.
```
Comment (hxrts) [inline docs/modules/group.md]:
```suggestion
A group is a set of accounts and associated weights. Group accounts may then be instantiated by creating a new decision policy. Each decision policy can have a different threshold for voting on and passing proposals.
```

File: docs/modules/group.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
+# Group Module
+
+**The `group` module facilitates the creation and management of multisig accounts and enables members to configure group policies that can submit and vote on specific message execution proposals.**
```
Comment (Pantani) [inline docs/modules/group.md]:
```suggestion
**The `group` module facilitates creating and managing multisig accounts and enables members to configure group policies that can submit and vote on specific message execution proposals.**
```

File: docs/modules/group.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+## Overview
+A group is a set of accounts and associated weights. Group accounts may then be instantiated by creating a new decision policy. Each decision policy can have a different threshold for voting on and passing proposals.
+
+See the following resources for additional information on using the `group` module.
+
+- [Cosmos SDK Group Module Docs](https://docs.cosmos.network/v0.46/modules/group/)
+- [Cosmos Developer Portal Group Tutorial](https://tutorials.cosmos.network/tutorials/understanding-group/)
+
+##### Group Administration
+Groups can be initialized with or without a decision policy. Groups must have an administrator account, but that account does not necessarily have to be a group member. The group may self-administer by setting the administrator to a group policy account. Upon creating a group, each member is assigned a weight that will correspond to voting power with respect to policy proposals.
```
Comment (Pantani) [inline docs/modules/group.md]:
```suggestion
Groups can be initialized with or without a decision policy. Groups must have an administrator account, but that account does not necessarily have to be a group member. The group may self-administer by setting the administrator to a group policy account. Upon creating a group, each member is assigned a weight corresponding to voting power concerning policy proposals.
```

File: docs/modules/group.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+The administrator can update group members, replace themselves, and update decision policies.
+
+##### Policies
+There are two types of decision policies, threshold and percentage decision policies. A threshold decision policy sets the minimum voting power cast in favor of a proposal for it to pass. A percentage decision policy requires a minimum percentage of voting power cast in favor of a policy in order for it to pass. Policies can be updated by a group admin and are the actual abstraction of multi-signature accounts.
+
+##### Proposals
+Proposals are submitted on behalf of an individual decision policy account and voted on. Proposals facilitate multiple arbitrary message execution as long as the decision policy account has the authority to execute them. For example, a decision policy account can't execute a bank send if it doesn't carry a balance.
```
Comment (Pantani) [inline docs/modules/group.md]:
```suggestion
Proposals are submitted on behalf of an individual decision policy account and voted on. Proposals facilitate multiple arbitrary message execution as long as the decision policy account has the authority to execute them. For example, a decision policy account can't perform a bank send if it doesn't carry a balance.
```

File: docs/modules/group.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    "min_execution_period": "0s",
+    "voting_period": "30s"
+  }
+}
+```
+
+### Submitting a Proposal
+Once a group has at least one decision policy, members can begin to submit proposals on behalf of a specific policy account. Run `gaiad tx group submit-proposal [proposal_json_file] [flags]` to submit a group proposal and see below for an example of the `proposal.json`
+
+> Note: Messages in the proposal can only be executed if the group policy account on the proposal has the authority to do so. In the below example, the policy account would have to have sufficient balance to execute the bank send message.
```
Comment (Pantani) [inline docs/modules/group.md]:
```suggestion
> Note: Messages in the proposal can only be executed if the group policy account on the proposal has the authority to do so. In the below example, the policy account would have sufficient balance to execute the bank send message.
```

File: docs/modules/group.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  "percentage": "0.5",
+  "windows": {
+    "min_execution_period": "0s",
+    "voting_period": "30s"
+  }
+}
+```
+
+### Submitting a Proposal
+Once a group has at least one decision policy, members can begin to submit proposals on behalf of a specific policy account. Run `gaiad tx group submit-proposal [proposal_json_file] [flags]` to submit a group proposal and see below for an example of the `proposal.json`
```
Comment (Pantani) [inline docs/modules/group.md]:
```suggestion
Once a group has at least one decision policy, members can begin to submit proposals on behalf of a specific policy account. Run `gaiad tx group submit-proposal [proposal_json_file] [flags]` to submit a group proposal, and see below for an example of the `proposal.json`
```

File: docs/modules/group.md (hunk 6 above)
Comment (glnro) [reply docs/modules/group.md]: A better way to write this might actually be `would need to have`

File: docs/modules/group.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+There are two types of decision policies, threshold and percentage decision policies. A threshold decision policy sets the minimum voting power cast in favor of a proposal for it to pass. A percentage decision policy requires a minimum percentage of voting power cast in favor of a policy in order for it to pass. Policies can be updated by a group admin and are the actual abstraction of multi-signature accounts.
+
+##### Proposals
+Proposals are submitted on behalf of an individual decision policy account and voted on. Proposals facilitate multiple arbitrary message execution as long as the decision policy account has the authority to execute them. For example, a decision policy account can't execute a bank send if it doesn't carry a balance.
+
+## Group Lifecycle
+
+**1. Group Creation**
+
+A group is initialized by the administrator. Both options exist to create the group with our without its first decision policy.
```
Comment (LexaMichaelides) [inline docs/modules/group.md]:
```suggestion
A group is initialized by the administrator. Both options exist to create the group with or without its first decision policy.
```

**Pull request merged**